plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TimestampCoercer.java

```diff
@@ -83,4 +89,64 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
                             toType));
         }
     }
+
+    public static class VarcharToShortTimestampCoercer
+            extends TypeCoercer<VarcharType, TimestampType>
+    {
+        public VarcharToShortTimestampCoercer(VarcharType fromType, TimestampType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            try {
+                Slice value = fromType.getSlice(block, position);
+                LocalDateTime dateTime = LOCAL_DATE_TIME.parse(value.toStringUtf8(), LocalDateTime::from);
+                long epochSecond = dateTime.toEpochSecond(UTC);
+                if (epochSecond < START_OF_MODERN_ERA_SECONDS) {
+                    throw new TrinoException(HIVE_INVALID_TIMESTAMP_COERCION, "Coercion on historical dates is not supported");
```

Comment:
Why we throw exception here, we are not able to adjust date using GregorianCalendar and `month -1 ` ?

Reply:
Hive supports different way of storing timestamp across various version (2 and 3)  and various types (ORC/Parquet/RC) for some it uses `java.sql.Timestamp` and other it uses a different way - so we are restricting them

Reply:
So we just cannot predict what is the way of storing such date in underlying datasource (is it for example shifted for -1 month or not), so we just deny such cases.

Reply:
It is not storing but actually the way it coerces from String to Timestamp